machine_learning_hep/processer.py

```diff
@@ -467,6 +465,8 @@ def applymodel(self, file_index):
             if self.p_mask_values:
                 mask_df(dfrecosk, self.p_mask_values)
             if self.doml is True:
+                from machine_learning_hep.models import \
```

Comment:
Good to import this only when actually needed, but wouldn't it be even better placing the conditional import in `steer_analysis.py`, so that it fails early in case the import is needed but fails?

Reply:
I don't think it is easily possible. The apply function is used inside the processer.py.
On the other hand, apply from models is imported at top level in the optimiser.py, which is imported in line 331 in steer analysis, if doml is True. That is before lines 372-375 where the (multi)processer apply function is invoked, which leads us to this apply import in processer.py, also conditional on doml.
So, I think, if there are any import problems, they will be revealed by line 331 in steer_analysis.py, so we don't need to re-check it.

Reply:
Ok, good for me then.